### PR TITLE
GDPR personal data retrieval

### PR DESCRIPTION
**Pull request by fenek**
This PR is a clean version of #2253.

It adds a CLI method for retrieving user's personal data.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[6366.1](https://travis-ci.org/esl/MongooseIM/jobs/524602836) / Erlang 19.3 / small_tests / 312261ed8b0c0dbdde16f5aebe81148468f1f649
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2285/6366/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2285/6366/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-04-25_19.03.43/lib.mongooseim.logs/run.2019-04-25_19.03.43/suite.log.html)

**Comment by mongoose-im** [comment]
[6373.1](https://travis-ci.org/esl/MongooseIM/jobs/524811317) / Erlang 19.3 / small_tests / 8e0af9273aed8552c62aa6aea70e7c4bba0ac6c0
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2285/6373/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2285/6373/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-04-26_08.00.16/lib.mongooseim.logs/run.2019-04-26_08.00.17/suite.log.html)

**Comment by mongoose-im** [comment]
[6379.1](https://travis-ci.org/esl/MongooseIM/jobs/524854125) / Erlang 19.3 / small_tests / 498dd480203db55f6ef0ba049ea66e0ac2bd7014
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2285/6379/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2285/6379/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-04-26_11.41.00/lib.mongooseim.logs/run.2019-04-26_11.41.01/suite.log.html)

File: src/admin_extra/service_admin_extra_gdpr.erl [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    FilePath = TmpDir ++ "/" ++ FileName,
+    Args = [FilePath, Username, Domain | FileList],
+    0 = run(Cmd, Args, ?CMD_TIMEOUT),
+    FileName.
+
+-spec get_logs_from_node(node(), gdpr:username(), gdpr:domain(), file:name()) -> file:name().
+get_logs_from_node(Node, Username, Domain, TmpDir) ->
+    {ok, ZippedData} = rpc:call(Node, ?MODULE, retrieve_logs, [Username, Domain]),
+    {ok, [File]} = zip:unzip(ZippedData, [{cwd, TmpDir}]),
+    string:prefix(File, TmpDir ++ "/").
```
Comment (ludwikbukowski) [inline src/admin_extra/service_admin_extra_gdpr.erl]: This won't work on Erlangs below 20

**Comment by mongoose-im** [comment]
[6395.1](https://travis-ci.org/esl/MongooseIM/jobs/528837375) / Erlang 19.3 / small_tests / a8c95cf86c0874599b57d7c92a3cb4e8c9b3c7d3
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2285/6395/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2285/6395/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-05-06_15.37.42/lib.mongooseim.logs/run.2019-05-06_15.37.43/suite.log.html)

**Comment by mongoose-im** [comment]
[6413.1](https://travis-ci.org/esl/MongooseIM/jobs/529264983) / Erlang 19.3 / small_tests / 117af257fb740ecd1df22af3641af4debca88655
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2285/6413/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2285/6413/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-05-07_15.45.04/lib.mongooseim.logs/run.2019-05-07_15.45.05/suite.log.html)

**Comment by mongoose-im** [comment]
[6430.1](https://travis-ci.org/esl/MongooseIM/jobs/529768668) / Erlang 19.3 / small_tests / e57dc1c65af94755c7ea9a481256920b3bb4fb4b
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2285/6430/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2285/6430/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-05-08_15.50.51/lib.mongooseim.logs/run.2019-05-08_15.50.52/suite.log.html)

**Comment by mongoose-im** [comment]
[6446.1](https://travis-ci.org/esl/MongooseIM/jobs/530234504) / Erlang 19.3 / small_tests / 74a3e0d0c4a5d1a60dac44a1cdd31f7113e6e868
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2285/6446/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2285/6446/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-05-09_12.15.42/lib.mongooseim.logs/run.2019-05-09_12.15.42/suite.log.html)

**Comment by mongoose-im** [comment]
[6461.1](https://travis-ci.org/esl/MongooseIM/jobs/530663119) / Erlang 19.3 / small_tests / 1a18f815e8eebb73a7fc7412cc7304fd6abc5efe
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2285/6461/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2285/6461/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-05-10_11.53.34/lib.mongooseim.logs/run.2019-05-10_11.53.35/suite.log.html)

**Comment by mongoose-im** [comment]
[6474.1](https://travis-ci.org/esl/MongooseIM/jobs/530867122) / Erlang 19.3 / small_tests / 1aecbed83018bfedeac0d56c85dbe7e9bf315d26
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2285/6474/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2285/6474/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-05-10_20.37.29/lib.mongooseim.logs/run.2019-05-10_20.37.30/suite.log.html)

**Comment by mongoose-im** [comment]
[6477.1](https://travis-ci.org/esl/MongooseIM/jobs/531346447) / Erlang 19.3 / small_tests / 1986460fc9b772b6eccaa7cc7d13ec71173a14b9
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2285/6477/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2285/6477/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-05-12_08.56.21/lib.mongooseim.logs/run.2019-05-12_08.56.22/suite.log.html)

**Comment by mongoose-im** [comment]
[6481.1](https://travis-ci.org/esl/MongooseIM/jobs/531701355) / Erlang 19.3 / small_tests / b8b7eb6703b68864ab01e734c966fb8f7c884b10
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2285/6481/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2285/6481/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-05-13_10.41.37/lib.mongooseim.logs/run.2019-05-13_10.41.38/suite.log.html)

**Pull request merged**